### PR TITLE
Workflow changes to fix old release binaries.

### PR DESCRIPTION
**Pull request by roshkhatri**
- Moves `build-config.json` to workflow dir to build old versions with new configs.
- Enables contributors to test release Wf on private repo by adding `github.event_name == 'workflow_dispatch' ||`

### CODE REVIEW

File: .github/workflows/build-release-packages.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
   # This job provides the version metadata from the tag for the other jobs to use.
   release-build-get-meta:
     name: Get metadata to build
-    if: github.repository == 'valkey-io/valkey'
+    if: github.event_name == 'workflow_dispatch' || github.repository == 'valkey-io/valkey'
```
Comment (zuiderkwast) [inline .github/workflows/build-release-packages.yml:26]: Is this to be able to test the workflow in a fork?

**Review by zuiderkwast — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/build-release-packages.yml (hunk 1 above)
Comment (roshkhatri) [reply .github/workflows/build-release-packages.yml:26]: Yes, contributors can use `workflow_dispatch` in their fork to trigger the workflow

Comment (roshkhatri) [reply .github/workflows/build-release-packages.yml:26]: Screenshot 2024-12-19 at 12 24 53 PM

Comment (zuiderkwast) [reply .github/workflows/build-release-packages.yml:26]: Thats what you wrote in the top comment, yey...

**Comment by zuiderkwast** [comment]
@enjoy-binbin @madolson FYI: #1429 only fixed the builds for future versions. This PR makes it possible to re-deploy binary releases using the workflow in the unstable branch, so we can fix the binary releases for the old versions.

**Pull request merged**